utils/utils.go

```diff
@@ -185,24 +185,35 @@ func pathToRegExp(localPath string) string {
 	if strings.HasSuffix(localPath, "/") || strings.HasSuffix(localPath, "\\") {
 		localPath += wildcard
 	}
-	localPath = "^" + localPath + "$"
-	return localPath
+	return "^" + localPath + "$"
 }
 
-// Replaces matched regular expression from path to corresponding {i} at target.
+// Replaces matched regular expression from path to corresponding placeholder {i} at target.
 // Example 1:
 //      pattern = "repoA/1(.*)234" ; path = "repoA/1hello234" ; target = "{1}" ; ignoreRepo = false
 //      returns "hello"
 // Example 2:
 //      pattern = "repoA/1(.*)234" ; path = "repoB/1hello234" ; target = "{1}" ; ignoreRepo = true
 //      returns "hello"
 func BuildTargetPath(pattern, path, target string, ignoreRepo bool) (string, error) {
-	if ignoreRepo {
+	asteriskIndex := strings.Index(pattern, "*")
+	slashIndex := strings.Index(pattern, "/")
+	if shouldRemoveRepo(ignoreRepo, asteriskIndex, slashIndex) {
+		// Required when working with virtual repositories as the pattern may contain the virtual-repository name,
```

Comment:
Required when ... --> Required the repository part of the path, when ...

Reply:
What do you think of the following:
`Removing the repository part of the path is required when working with virtual repositories, as the pattern may contain the virtual-repository name, but the path contains the local-repository name.`

Reply:
LGTM.